### PR TITLE
#17 Adding CRUD Design , functions in category page

### PR DESCRIPTION
**Pull request by Nada257**

### CODE REVIEW

File: retail-managment-system/rms-ui/angular.json [hunk 1]
```diff
@@ -1,5 +1,8 @@
 {
   "$schema": "./node_modules/@angular/cli/lib/config/schema.json",
+  "cli": {
+    "analytics": "6848b278-fc38-4e10-9d6f-66c37dbe2b18"
+  },
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/angular.json:5]: delete this please

File: retail-managment-system/rms-ui/src/index.html [hunk 2]
```diff
@@ -5,9 +5,8 @@
     <title>Rms</title>
     <base href="/" />
     <meta name="viewport" content="width=device-width, initial-scale=1" />
-    <link rel="icon" type="image/x-icon" href="favicon.ico" />
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/index.html:8]: why delete it?

File: retail-managment-system/rms-ui/src/index.html [hunk 3]
```diff
@@ -5,9 +5,8 @@
...
   </head>
-  <body>
+  <body class="mat-typography">
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/index.html:9]: why that?

File: retail-managment-system/rms-ui/src/styles.scss [hunk 4]
```diff
@@ -1,3 +1,20 @@
...
+$rms-ui-warn: mat.define-palette(mat.$red-palette);
+
+$rms-ui-theme: mat.define-light-theme((
+  color: (
+    primary: $rms-ui-primary,
+    accent: $rms-ui-accent,
+    warn: $rms-ui-warn,
+  )
+));
+@include mat.all-component-themes($rms-ui-theme);
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/styles.scss:17]: why add this lines from line 1 to 17?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+  Validators,
+} from '@angular/forms';
+import { Component, OnInit } from '@angular/core';
+import { Category } from 'src/app/domain/category/models/category';
+import { PageEvent } from '@angular/material/paginator';
+
+@Component({
+  selector: 'app-category-page',
+  templateUrl: './category-page.component.html',
+  styleUrls: ['./category-page.component.scss'],
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:15]: if you unused css file delete the file.

File: retail-managment-system/rms-ui/src/app/core/services/resource.service.ts [hunk 6]
```diff
@@ -73,17 +73,26 @@ export abstract class ResourceService<T> {
...
+  // update(resource: T): Observable<any> {
+  //   // @ts-ignore
+  //   return this.httpClient
+  //     .put(`${this.APIUrl}/${resource.id}`, this.toServerModel(resource))
+  //     .pipe(
+  //       catchError((err) => {
+  //         throw new Error(err.message);
+  //       })
+  //     );
+  // }
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/core/services/resource.service.ts:85]: don't push comments.

File: retail-managment-system/rms-ui/src/app/domain/category/category.repository.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+    if (!entity.id) {
+      return {
+        name: entity.name,
+        status: entity.status,
+      };
+    } else {
+      return {
+        id: entity.id,
+        name: entity.name,
+        status: entity.status,
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/domain/category/category.repository.ts:23]: why you added to server model here?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 8]
```diff
@@ -0,0 +1,96 @@
...
+      this.getAllCategory();
+    });
+  }
+  //update all value
+  UpdateCategory(): void {
+    this.categoryRepository.update(this.categoryForm.value).subscribe(() => {
+      this.getAllCategory();
+    });
+  }
+  pageChanged(event: PageEvent): void {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:64]: why you add this function?

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html [hunk 9]
```diff
@@ -0,0 +1,307 @@
...
+                    matInput
+                    placeholder="enter your name"
+                    fxFlex="60"
+                  />
+                </div>
+              </div>
+              <div class="col-lg-4">
+                <div class="mb-3">
+                  <input
+                    formControlName="id"
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html:207]: id add from backend not frontend

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html [hunk 10]
```diff
@@ -0,0 +1,307 @@
...
+                  <th>ID</th>
+                  <th></th>
+                </tr>
+              </thead>
+              <tbody>
+                <tr *ngFor="let categori of allCategory; index as i">
+                  <td>{{ i + 1 }}</td>
+                  <td>{{ categori.name }}</td>
+                  <td>{{ categori.status }}</td>
+                  <td>{{ categori.id }}</td>
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.html:276]: don't show id in table

File: retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts [hunk 11]
```diff
@@ -0,0 +1,96 @@
...
+  //Get all data
+  getAllCategory(): void {
+    this.categoryRepository.getList().subscribe((result: any) => {
+      this.allCategory = result;
+      // this.totalRows = result.pagination.itemCount;
+      console.log(this.allCategory);
+    });
+  }
+
+  onSubmit() {
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/category-page.component.ts:78]: this function unused

**Review by ZinabHashem — CHANGES_REQUESTED** [review]

File: retail-managment-system/rms-ui/src/index.html (hunk 2 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/index.html:8]: Not using it anymore

File: retail-managment-system/rms-ui/src/styles.scss (hunk 4 above)
Comment (Nada257) [reply retail-managment-system/rms-ui/src/styles.scss:17]: Using it to style the table with angular material

Comment (ZinabHashem) [reply retail-managment-system/rms-ui/src/styles.scss:17]: without it style of table not work?

File: retail-managment-system/rms-ui/src/app/pages/category-page/pop-up/pop-up.component.ts [hunk 12]
```diff
@@ -20,23 +15,15 @@ export class PopUpComponent implements OnInit {
...
     this.getAllCategory();
     this.fetchData(this.data);
   }
-  categoriForm() {
-    this.popupForm = this.formBuilder.group({
-      id: [''],
-      name: ['', [Validators.required]],
-      status: ['', [Validators.required]],
-    });
-  }
```
Comment (ZinabHashem) [inline retail-managment-system/rms-ui/src/app/pages/category-page/pop-up/pop-up.component.ts]: why you delete this form ?!

**Comment by ZinabHashem** [comment]
please make popup auto close when add or update

**Pull request merged**